frontend_lib/src/translation.js

```diff
@@ -0,0 +1,44 @@
+import i18n from 'i18next'
+
+export const TRANSLATION_STATE = {
+  DISABLED: 'disabled',
+  UNTRANSLATED: 'untranslated',
+  PENDING: 'pending',
+  TRANSLATED: 'translated'
+}
+
+export const getTranslationApiErrorMessage = (response) => {
+  const errors = [
+    { status: 502, message: i18n.t('Translation is not available') },
+    { status: 504, message: i18n.t('Translation is not available') }
+  ]
+  if (!response.ok) {
+    const error = errors.find(e => e.status === response.status) || { status: null, message: i18n.t('Unknown error') }
+    return i18n.t(
+      'Error while translating the document: {{details}}',
+      { details: error.message }
+    )
+  }
+  return null
+}
+
+export const setTranslatedRawContent = (object, translatedRawContent) => {
+  return {
+    ...object,
+    translatedRawContent,
+    translationState: TRANSLATION_STATE.TRANSLATED
+  }
+}
+
+export const setTranslationState = (object, translationState) => {
+  return {
+    ...object,
+    translationState
+  }
+}
```

Comment:
I'm a bit confused by these 2 functions, the first one is because it's only used once and the the second one because it doesn't do a lot of things.

But I'm assuming their main purpose is to keep appContentFactory agnostic of the properties the setState has to update?

Reply:
In fact I was planning to use them in the html document translation mechanism too, but I didn't in the end… I probably could remove them.